### PR TITLE
(FACT-2361) ZPool for Solaris

### PR DESCRIPTION
**Pull request by Filipovici-Andrei**
Implemented ZPool version and feature numbers facts.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,3 +54,4 @@ build-iPhoneSimulator/
 
 .rspec_status
 Gemfile.lock
+Gemfile.local
```
Comment (IrimieBogdan) [inline .gitignore]: This is duplicated and should be removed.

File: spec/facter/facts/solaris/zfs_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      expected_fact = double(Facter::ResolvedFact, name: 'zfs_featurenumbers', value: '1,2,3,4,5')
+      allow(Facter::Resolvers::Solaris::ZFS).to receive(:resolve).with(:zfs_featurenumbers).and_return('1,2,3,4,5')
+      allow(Facter::ResolvedFact).to receive(:new).with('zfs_featurenumbers', '1,2,3,4,5').and_return(expected_fact)
+
+      fact = Facter::Solaris::ZFSFeatureNumbers.new
+
+      expect(fact.call_the_resolver).to eq(expected_fact)
+    end
+  end
+end
```
Comment (IrimieBogdan) [inline spec/facter/facts/solaris/zfs_spec.rb]: The test should be written fallowing the new pattern https://github.com/puppetlabs/facter-ng/compare/sample_spec?expand=1

File: spec/facter/facts/solaris/zpool_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      expected_fact = double(Facter::ResolvedFact, name: 'zpool_featurenumbers', value: '1,2,3,4,5,6,7')
+      allow(Facter::Resolvers::Solaris::ZPool).to receive(:resolve)
+        .with('zpool_featurenumbers')
+        .and_return('1,2,3,4,5,6,7')
+      allow(Facter::ResolvedFact).to receive(:new)
+        .with('zpool_feature', '1,2,3,4,5,6,7')
+        .and_return(expected_fact)
+    end
+  end
+end
```
Comment (IrimieBogdan) [inline spec/facter/facts/solaris/zpool_spec.rb]: The test should be written fallowing the new pattern https://github.com/puppetlabs/facter-ng/compare/sample_spec?expand=1

File: spec/facter/resolvers/solaris/zfs_resolver_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+describe 'SolarisZFS' do
+  before do
+    status = double(Process::Status, to_s: st)
+    expect(Open3).to receive(:capture2)
+      .with('zfs upgrade -v')
+      .ordered
+      .and_return([output, status])
+  end
+
+  after do
```
Comment (IrimieBogdan) [inline spec/facter/resolvers/solaris/zfs_resolver_spec.rb]: This is ok, it is equivalent to `after(:each)`

**Comment by gimmyxd** [comment]
Something looks off, this has a lot of commits coming from other branches. Did you rebase with master?

File: .gitignore (hunk 1 above)
Comment (Filipovici-Andrei) [reply .gitignore]: It's lock and local. The same as in master.

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**